### PR TITLE
[PIN-2616] Solving bug related to consumer not allowed to update agreement in draft

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 1]
```diff
@@ -204,8 +204,7 @@ final case class AgreementApiServiceImpl(
       requesterOrgId <- getOrganizationIdFutureUUID(contexts)
       agreementUUID  <- agreementId.toFutureUUID
       agreement      <- agreementManagementService.getAgreementById(agreementUUID)
-      _              <- assertRequesterIsProducer(requesterOrgId, agreement)
-      // Correct way to check if agreement is in draft state?
+      _              <- assertRequesterIsConsumer(requesterOrgId, agreement)
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:207]: Are you sure that a producer should never call this API? Maybe we should consider using the condition `assertRequesterIsConsumer(requesterOrgId, agreement) || assertRequesterIsProducer(requesterOrgId, agreement)`, or maybe we should consider the chance to let the producer and the consumer update an agreement differently.

Comment (bloodsky) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:207]: Maybe we can use `assertRequesterIsConsumerOrProducer()` that is already present. From a talk with "frontend" only consumers needed to do this operation. @galales @beetlecrunch

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:207]: Just be sure with frontend that this API doesn't get called in some other UX operation by a producer 👍

Comment (galales) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:207]: I can confirm that this api should not be called by the producer 👍

**Review by galales — APPROVED** [review]
🚀

**Review by TonioGela — APPROVED** [review]

**Pull request merged**